### PR TITLE
Plfm 2038

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/wiki/WikiPage.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (john-hill) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/wiki/WikiPage.json:1]: Need a better top level description

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java [hunk 2]
```diff
@@ -29,43 +29,101 @@
...
+ * <p>
+ * These services provide provides support for creating, reading, updating, and
+ * deleting (CRUD) the WikiPage model objects.
+ * </p>
+ * <p>
+ * WikiPages are composed of two major parts; the raw markdown text and a list
+ * of file attachments. For example, to embed an image from an end-user's
+ * machine into a WikiPage, the image file must first be upload to Synapse as <a
+ * href="${org.sagebionetworks.repo.model.file.FileHandle}">FileHandle</a> (see
+ * File Services). The FileHandle ID can then be added to a
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java]: need  a link the file services.

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java [hunk 3]
```diff
@@ -29,43 +29,101 @@
...
+ * WikiPage.attachmentFileHandleIds list. See <a
+ * href="https://www.synapse.org">www.synapse.org</a> for details on the
+ * supported markdown syntax.
+ * </p>
+ * 
+ * <p>
+ * WikiPages are not stand-alone objects, instead they are a component of
+ * another object such as an Entity, Competition or Evaluation. For example,
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java]: no such thing as a competition

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java [hunk 4]
```diff
@@ -29,43 +29,101 @@
...
+ * when a WikiPage is created for an Entity, the Entity becomes the "owner" of
+ * the WikiPage. Access to the WikiPage is always tied to its owner. For
+ * example, to GET a WikiPage of an Entity, the caller must have read permission
+ * on the Entity.
+ * 
+ * Note: WikiPages can be nested to created a hierarchy of sub-pages. However,
+ * there can only be one root WikiPage per owner object, and all sub-pages are
+ * considered to be owned by the same object as the root page.
+ * </p>
  */
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java:75]: Add a section about the wiki tree for navigation.

**Comment by john-hill** [comment]
On entity delete mention that recursive delete and also point them to the recycle feature for recoverable deletes.

**Comment by john-hill** [comment]
WikiHeader JSON needs a top level description.

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java [hunk 5]
```diff
@@ -243,96 +487,276 @@ void deleteCompetitionWikiPage(
...
+			NotFoundException {
+		return serviceProvider.getWikiService().getWikiHeaderTree(userId,
+				ownerId, ObjectType.EVALUATION, limit, offset);
 	}
+
 	// Handles
+	/**
+	 * Get the list of FileHandles for a all file attachments of a specific
+	 * WikiPage for a given owning Entity. This list will include Previews if
+	 * they exist and will provides information about file sizes, content types
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java]: provide

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java [hunk 6]
```diff
@@ -243,96 +487,276 @@ void deleteCompetitionWikiPage(
...
-		return serviceProvider.getWikiService().getAttachmentFileHandles(userId, new WikiPageKey(ownerId, ObjectType.ENTITY, wikiId));
+		return serviceProvider.getWikiService().getAttachmentFileHandles(
+				userId, new WikiPageKey(ownerId, ObjectType.ENTITY, wikiId));
 	}
-	
+
+	/**
+	 * Get the list of FileHandles for a all file attachments of a specific
+	 * WikiPage for a given owning Evaluation. This list will include Previews
+	 * if they exist and will provides information about file sizes, content
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java]: provide

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java [hunk 7]
```diff
@@ -29,43 +29,101 @@
...
+ * <p>
+ * The Synapse <a
+ * href="${org.sagebionetworks.repo.model.wiki.WikiPage}">WikiPage</a> model
+ * object contains the data needed to render an end-user crafted web page. The
+ * Synapse Web Client will dynamically render a WikiPage into a combination of
+ * HTML, CSS and Javascript which is then finally rendered as a web page in the
+ * client's web browser.
+ * </p>
+ * <p>
+ * These services provide provides support for creating, reading, updating, and
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java]: duplicate provide

**Comment by john-hill** [comment]
FileHandleResults schema needs a description

File: services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java [hunk 8]
```diff
@@ -243,96 +487,276 @@ void deleteCompetitionWikiPage(
...
+	 */
 	@RequestMapping(value = UrlHelpers.EVALUATION_WIKI_ID_ATTCHMENT_FILE_PREVIEW, method = RequestMethod.GET)
 	public @ResponseBody
 	void getCompetitionAttachmenthPreviewFile(
 			@RequestParam(value = AuthorizationConstants.USER_ID_PARAM, required = false) String userId,
-			@PathVariable String ownerId,
-			@PathVariable String wikiId,
+			@PathVariable String ownerId, @PathVariable String wikiId,
 			@RequestParam String fileName,
-			@RequestParam (required = false) Boolean redirect,
```
Comment (john-hill) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/controller/WikiController.java:329]: fileName is required

**Comment by john-hill** [comment]
for Request Object and Response Object is null then use "None"

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/wiki/WikiPage.json [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {
-	"description": "JSON schema for WikiPage",
+	"description": "The WikiPage model object represent a single user managed wiki page.",
```
Comment (brucehoff) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/wiki/WikiPage.json:2]: "representS"

**Comment by brucehoff** [comment]
awesome job!!

**Pull request merged**